pkg/utils/kubernetes/health/health.go

```diff
@@ -248,8 +247,8 @@ func CheckSeed(seed *gardencorev1beta1.Seed, identity *gardencorev1beta1.Gardene
 	if seed.Status.ObservedGeneration < seed.Generation {
 		return fmt.Errorf("observed generation outdated (%d/%d)", seed.Status.ObservedGeneration, seed.Generation)
 	}
-	if !apiequality.Semantic.DeepEqual(seed.Status.Gardener, identity) {
-		return fmt.Errorf("observing Gardener version not up to date (%v/%v)", seed.Status.Gardener, identity)
+	if seed.Status.Gardener.Version != identity.Version {
```

Comment:
Instead of removing it always, should we rather have a special check for cp migration that only considers the version? Otherwise we miss cases in which the gardenlet pod is redeployed and needs to bootstrap the seed again.

Reply:
Agreed, it must not be removed in principle. @kris94 In our initial discussion I suggested adding another function `CheckSeedForMigration` without this check, or adding a boolean parameter to this function e.g. `forMigration`, and disabling the check only when it's true.